✨ scaffold fleetconfig v1beta1 resources and webhooks

arturshadnik opened the pull request:
## Description

- Adds a `PROJECT` file to the fleetconfig-controller project to track created resources and allow `kubebuilder` to function
- Moves the FleetConfig reconciler into a `v1alpha1` module to organize the controllers
- Scaffolds 2 new `v1beta1` resources, `Hub` and `Spoke` in the same `fleetconfig.open-cluster-management.io` group.
- Generates new CRDs, related `config/` resources.
- Updates Helm chart `manager-rbac` template to provide the same permissions for the new resources as for FleetConfig
- Adds a helper to find the root project directory from different test directories.

No implementation on this PR, so that it can be review in a followup with no noise.

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- New Features
  - Adds cluster-scoped Hub and Spoke (v1beta1) CRDs, controllers, and admission webhooks (defaulting/validation); Helm chart now includes Hub/Spoke CRDs.
- Tests
  - Adds envtest-based controller and webhook test suites for v1beta1.
- Refactor
  - Adopts multi-version layout (v1alpha1/v1beta1) with updated scheme registration and RBAC expanded for hubs/spokes and their status.
- Chores
  - Simplifies manifest/CRD generation pipeline, removes legacy kustomize/webhook patches and sample manifests, and adds project scaffolding and generated deepcopy code.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

arturshadnik commented on fleetconfig-controller/api/v1beta1/hub_types.go:
```diff
@@ -0,0 +1,76 @@
+type Hub struct {
+	metav1.TypeMeta `json:",inline"`
+
+	// metadata is a standard object metadata
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty,omitzero"`
+
+	// spec defines the desired state of Hub
+	// +required
+	Spec HubSpec `json:"spec"`
+
+	// status defines the observed state of Hub
+	// +optional
+	Status HubStatus `json:"status,omitempty,omitzero"`
+}
```
https://pkg.go.dev/github.com/go-json-experiment/json

> omitzero: When marshaling, the "omitzero" option specifies that the struct field should be omitted if the field value is zero as determined by the "IsZero() bool" method if present, otherwise based on whether the field is the zero Go value. This option has no effect when unmarshaling.

TylerGillson reviewed (commented):
We don't actually use anything under `config/`. Can you please remove that dir entirely and update the controller-gen marker to output the CRDs directly into the helm chart's `crds` dir?

arturshadnik commented on fleetconfig-controller/internal/controller/v1beta1/spoke_controller_test.go:
```diff
@@ -0,0 +1,84 @@
+			By("creating the custom resource for the Kind Spoke")
+			err := k8sClient.Get(ctx, typeNamespacedName, spoke)
+			if err != nil && errors.IsNotFound(err) {
+				resource := &v1beta1.Spoke{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      resourceName,
+						Namespace: "default",
+					},
+					// TODO(user): Specify other spec details if needed.
+				}
+				Expect(k8sClient.Create(ctx, resource)).To(Succeed())
```
The tests dont fail tho, so there is no issue.

arturshadnik commented on fleetconfig-controller/internal/webhook/v1beta1/hub_webhook_test.go:
```diff
@@ -0,0 +1,88 @@
+	var (
+		obj       *fleetconfigopenclustermanagementiov1beta1.Hub
+		oldObj    *fleetconfigopenclustermanagementiov1beta1.Hub
+		validator HubCustomValidator
+		defaulter HubCustomDefaulter
+	)
+
+	BeforeEach(func() {
+		obj = &fleetconfigopenclustermanagementiov1beta1.Hub{}
+		oldObj = &fleetconfigopenclustermanagementiov1beta1.Hub{}
+		validator = HubCustomValidator{}
+		Expect(validator).NotTo(BeNil(), "Expected validator to be initialized")
+		defaulter = HubCustomDefaulter{}
+		Expect(defaulter).NotTo(BeNil(), "Expected defaulter to be initialized")
+		Expect(oldObj).NotTo(BeNil(), "Expected oldObj to be initialized")
+		Expect(obj).NotTo(BeNil(), "Expected obj to be initialized")
```
the tests are passing so there is no issue

arturshadnik commented:
> @TylerGillson we do have a `devspace` manifest under there, but I removed everything else and updated the make target

I's also like to leave in the `webhook` config. It's used to install webhooks in integration tests, and it'll be easier to maintain this way.
